### PR TITLE
fix :beetle:'s with current-user service

### PR DESCRIPTION
**Pull request by vevix**
Fixes issues with new current user injection and saving data, also cleans up the initializer.

### CODE REVIEW

File: frontend/app/initializers/current-user.js [hunk 1]
```diff
@@ -1,19 +1,13 @@
-import currentUserProxy from '../services/current-user.js';
-
 export function initialize(container, app) {
-  var user, store = container.lookup('store:main');
-
-  app.register('service:current-user', currentUserProxy);
```
Comment (NuckChorris) [inline frontend/app/initializers/current-user.js]: Don't we still need to register it into the container?

Comment (vevix) [inline frontend/app/initializers/current-user.js]: ember-cli will handle that

Comment (NuckChorris) [inline frontend/app/initializers/current-user.js]: I thought that was a later addition in ember-cli, around Ember 1.10?

Comment (vevix) [inline frontend/app/initializers/current-user.js]: They made it so you can inject services into objects around that version with syntax like:
`currentUser: Ember.service.inject()`

but this is just ember-resolver magic.

**Comment by NuckChorris** [comment]
Sweet, LGTM

**Pull request merged**